registry/config.go

```diff
@@ -99,8 +99,12 @@ func newServiceConfig(options ServiceOptions) *serviceConfig {
 	}
 
 	config.LoadAllowNondistributableArtifacts(options.AllowNondistributableArtifacts)
-	config.LoadMirrors(options.Mirrors)
-	config.LoadInsecureRegistries(options.InsecureRegistries)
+	if err := config.LoadMirrors(options.Mirrors); err != nil {
+		logrus.Fatalf("invalid registry mirror configurations: %s", err)
```

Comment:
Yes I think that would be better; I also think we can drop the `invalid registry mirror configurations` and `invalid insecure registry configurations` prefix here; the error itself looks pretty self-describing already;

```
invalid mirror: unsupported scheme "example.com" in "example.com:5000
```

When handling in the daemon, we could have a prefix added, similar as what's used for _reloading_ the configuration, e.g.:

```
Error configuring the daemon: 
```